### PR TITLE
40-0247 - Refactor Vet Supporting Doc page file-upload

### PR DESCRIPTION
**Pull request by tlei123**
## Summary

-  Refactor PMC (40-0247) Veteran Supporting Documentation page:
    - Add missing uploaded filename display
    - Modularize functions out of page uiSchema
    - Add unit-tests for modularized functions
    - Add H3 to page content
- Team: Veteran Facing Forms
- Flipper not implemented yet

## Related issue(s)

- _Links to tickets created in va.gov-team-forms repo_
department-of-veterans-affairs/va.gov-team-forms#585
department-of-veterans-affairs/va.gov-team-forms#648
department-of-veterans-affairs/va.gov-team-forms#662

## Testing done

- Local browser-testing and updated unit-tests were successful

## What areas of the site does it impact?

This form ONLY

### CODE REVIEW

File: src/applications/simple-forms/40-0247/tests/e2e/fixtures/mocks/local-mock-api-reponses.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/* eslint-disable camelcase */
```
[the remark this answers is not in the document]
Comment (tlei123) [reply src/applications/simple-forms/40-0247/tests/e2e/fixtures/mocks/local-mock-api-reponses.js:1]: Yes, most of the response keys are camelCase, so we need this rule disabled for the whole file.  There's an enable at the bottom of the file to turn the rule back on.

File: src/applications/simple-forms/40-0247/pages/veteranSupportDocs.js [hunk 2]
```diff
@@ -4,32 +4,21 @@ import environment from 'platform/utilities/environment';
...
-          </a>
-        </p>
-        <p className="vads-u-margin-bottom--4">
-          We prefer that you upload the Veteran’s or Reservist’s DD214.
-        </p>
-      </>
+    'ui:title': (
+      <h3 className="vads-u-margin-y--0">
+        Upload documents (preferably DD214)
+      </h3>
```
Comment (ndsprinkle) [inline src/applications/simple-forms/40-0247/pages/veteranSupportDocs.js:19]: `titleUI` here? https://github.com/department-of-veterans-affairs/vets-website/blob/main/src/platform/forms-system/src/js/web-component-patterns/titlePattern.js#L23

**Comment by tlei123** [comment]
@rhasselle-oddball , [follow-up ticket #673](https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/673) opened to investigate encrypted-PDF-password functionality.

File: src/applications/simple-forms/40-0247/pages/veteranSupportDocs.js (hunk 2 above)
Comment (tlei123) [reply src/applications/simple-forms/40-0247/pages/veteranSupportDocs.js]: Refactored.  Thanks for catching this!

File: src/applications/simple-forms/40-0247/helpers.js [hunk 3]
```diff
@@ -1,5 +1,65 @@
...
+    }
+  };
+
+  setTimeout(() => {
+    focusFileCard();
+  }, 100);
+
+  return {
+    name,
+    confirmationCode: data.attributes.guid,
```
Comment (Thrillberg) [inline src/applications/simple-forms/40-0247/helpers.js:63]: The BE returns an object with key `confirmation_code`, not `guid`. [This line in the test file shows the keys it returns (`confirmation_code`, `name`, and `size`)](https://github.com/department-of-veterans-affairs/vets-api/blob/231fab78cc4a4ef163f8082f4c0cfcc21b189055/modules/simple_forms_api/spec/requests/v1/uploads_spec.rb#L83).

So I think this should be `confirmationCode: data.attributes.confirmation_code`?

**Review by Thrillberg — APPROVED** [review]
Just one small thing, but otherwise looks great to me! The API URL is also not right but I think it might make the most sense to merge this as-is and fix it everywhere it needs fixing as a separate small PR. What do you think? (This is the actual path we'll use, I think: `v0/simple_forms/submit_supporting_documents`)

File: src/applications/simple-forms/40-0247/helpers.js (hunk 3 above)
Comment (tlei123) [reply src/applications/simple-forms/40-0247/helpers.js]: I just fixed this data-reference in helpers.js, as well as my mock-data keys.

**Review by ataker — APPROVED** [review]

**Pull request merged**